sbt-scalajs-bundler/src/main/scala/scalajsbundler/BundlerFile.scala

```diff
@@ -65,22 +65,25 @@ object BundlerFile {
       */
     def asLibrary(stats: Option[WebpackStats]): Library =
       Library(project,
-              targetDir
-                .resolve(stats.flatMap(_.assetName(project)).fold(Library.fileName(project))(identity))
-                .toFile,
-              stats.map(_.assets.map(a => targetDir.resolve(a.name).toFile)).getOrElse(Nil))
+              stats.flatMap { s =>
+                s.resolveAsset(targetDir, project)
+              }.getOrElse(targetDir.resolve(Library.fileName(project)).toFile),
```

Comment:
Do you mean to use `Library.fileName(project)` instead of `project` in line 69? In line 70 we don't have access to `stats`

Reply:
I dug into this some more, and the current implementation is indeed correct, it's just that webpack 3 doesn't write `outputPath` to stats either, only webpack 4 does. In v3 there doesn't seem to be a config to add outputPath to stats either.

Reply:
I have mostly tested on wbpack 4, though there is a test for Webpack 4. Indeed there are differences on stats across versión and they are not documented

However, the `getOrElse` route should be used for that case, it may well be that we need adjustments for that route

Reply:
Sorry, there is a single test for Webpack 2. It has triggered issues on several occasions.